### PR TITLE
feat(ESSNTL-5302): Add delete staleness endpoint

### PR DESCRIPTION
**Pull request by fhbash**
# Overview

This PR is being created to address [ESSNTL-5302](https://issues.redhat.com/browse/ESSNTL-5302).
- Adding `delete` endpoint for Account Staleness endpoint
- Update openapi schema file to reflect new endpoint
- 
## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by jpramos123 — DISMISSED** [review]
LGTM

File: tests/fixtures/api_fixtures.py [hunk 1]
```diff
@@ -180,3 +180,11 @@ def _api_create_account_staleness(
         return do_request(flask_client.post, STALENESS_URL, identity, staleness_data, query_parameters, extra_headers)
 
     return _api_create_account_staleness
+
+
+@pytest.fixture(scope="function")
+def api_delete_account_staleness(flask_client):
+    def _api_delete_account_staleness(identity=USER_IDENTITY, query_parameters=None, extra_headers=None):
+        return do_request(flask_client.delete, STALENESS_URL, identity, query_parameters, extra_headers)
```
Comment (kruai) [inline tests/fixtures/api_fixtures.py:188]: Does it make sense to use this function with query parameters or extra headers? It's fine as-is, just wondering bc it doesn't look like any query params are allowed on that endpoint, and those are both optional params on `do_request`

**Review by kruai — CHANGES_REQUESTED** [review]
This should have at least one RBAC test, but it looks good besides that

File: tests/test_api_account_staleness_delete.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    response_status, response_data = api_delete_account_staleness()
+    assert_response_status(response_status, 204)
+
+    # checking if the record was really removed
+    deleted_staleness = db_get_account_staleness_culling(saved_staleness.org_id)
+    assert deleted_staleness is None
+
+
+def test_delete_non_existing_account_staleness(api_delete_account_staleness):
```
Comment (thearifismail) [inline tests/test_api_account_staleness_delete.py:24]: the test should check for NoResultFound exception

**Review by thearifismail — COMMENTED** [review]
One minor comment for this PR.

File: tests/test_api_account_staleness_delete.py (hunk 2 above)
Comment (fellipeh) [reply tests/test_api_account_staleness_delete.py:24]: This is an API test, `NoResultFound` is encapsulated into an `HTTP_404_NOT_FOUND` http response, so, afaik there isn't possible to check the exception here.

File: api/account_staleness.py [hunk 3]
```diff
@@ -69,6 +72,17 @@ def create_staleness(body):
     return flask_json_response(serialize_account_staleness_response(created_staleness), status.HTTP_201_CREATED)
 
 
+@api_operation
```
Comment (jpramos123) [inline api/account_staleness.py:75]: Lets start applying the feature flags:

Use this flag: "hbi.custom-staleness"

See this as a reference: https://github.com/RedHatInsights/insights-host-inventory/blob/master/api/group.py#L69

**Review by jpramos123 — CHANGES_REQUESTED** [review]
Lets add the feature flag

File: tests/test_api_account_staleness_delete.py (hunk 2 above)
Comment (kruai) [reply tests/test_api_account_staleness_delete.py:27]: I agree - NoResultFound is what results in the 404, so I'd argue that's what it's testing

**Review by jpramos123 — APPROVED** [review]
LGTM

**Review by thearifismail — APPROVED** [review]
To address the confusing naming concerns, here is the Jira card [ESSNTL-5336](https://issues.redhat.com/browse/ESSNTL-5336)

**Pull request merged**